### PR TITLE
refactor(component-testing): simplify vue mount

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes CT-367

### User facing changelog

Fixes some of the memory leaks in with `@cypress/vue`

### Additional details
- Avoid recreating the root element at each test
- Make sure we properly destroy the vue instance after each iteration

This might fix some issues with devtools

### PR Tasks

- [x] Has the original issue or this PR been tagged with a release in ZenHub?

### CODE REVIEW

File: npm/vue/src/index.ts [hunk 1]
```diff
@@ -63,7 +64,7 @@ const installMixins = (Vue, options) => {
   }
 }
 
-const hasStore = ({ store }: { store: any }) => store && store._vm // @ts-ignore
+const hasStore = ({ store }: { store: any }): boolean => store && store._vm
```
Comment (dmtrKovalenko) [inline npm/vue/src/index.ts:67]:
```suggestion
const hasStore = ({ store }: { store: any }) => Boolean(store && store._vm)
```

File: npm/vue/src/index.ts [hunk 2]
```diff
@@ -357,14 +358,18 @@ export const mount = (
...
     let el = document.getElementById(ROOT_ID)
 
-    // If the target div doesn't exist, create it
-    if (!el) {
-      el = renderTestingPlatform(document.head.innerHTML)
+    // first make sure the previous Vue instance
+    // no longer exists
+    if (Cypress.vue) {
+      Cypress.vue.$destroy()
+      delete Cypress.vue
```
Comment (dmtrKovalenko) [inline npm/vue/src/index.ts:367]: Do we really need to run `delete`? Using delete keyword (especially for reused giant objects like Cypress) can lead to deoptimizations of v8. Maybe we can use `= undefined` instead? 

> some source https://stackoverflow.com/questions/59431589/javascript-performance-constraints-of-delete-keyword

Comment (elevatebart) [reply npm/vue/src/index.ts:367]: I agree. 
We don't even need to run the delete. 
I was thinking of avoiding to run `$destroy()` on an empty object but the overhead would be fairly small

The other good option would be if there was a `destroyed` flag on the vm instance, then we could condition on that.

I will delete the delete line for now.

Comment (elevatebart) [reply npm/vue/src/index.ts]: removed delete line

**Review by dmtrKovalenko — APPROVED** [review]

**Comment by lmiller1990** [comment]
We could use the feature that is built in to test utils: https://vue-test-utils.vuejs.org/api/#enableautodestroy-hook

**Review by lmiller1990 — CHANGES_REQUESTED** [review]
I think we could consider using the version built into test utils: We could use the feature that is built in to test utils: https://vue-test-utils.vuejs.org/api/#enableautodestroy-hook

**Comment by elevatebart** [comment]
@lmiller1990 I do not think we can put anything in afterEach without reaching for a support file.
It would uselessly complicate the installation process.

Where were you thinking we call this `enableAutoDestroy` function?

**Comment by elevatebart** [comment]
I understand after looking at your vite PR what you mean. 
Tell me if my implementation seems fine.

File: npm/vue/src/index.ts [hunk 3]
```diff
@@ -305,6 +307,8 @@ function failTestOnVueError (err, vm, info) {
   window.top.onerror(err)
 }
 
+enableAutoDestroy(beforeEach)
```
Comment (lmiller1990) [inline npm/vue/src/index.ts:310]: This is what I was thinking - although based on my new understanding we should be using cypress events. I think this is okay for now!

**Review by lmiller1990 — APPROVED** [review]

**Review by JessicaSachs — DISMISSED** [review]
This looks okay to me. It's an improvement on where we currently are. There are some fundamental issues with how we are cleaning up, but those will be addressed when we refactor these to use Cypress's internal events.

**Review by lmiller1990 — APPROVED** [review]
✅

**Review by JessicaSachs — APPROVED** [review]
Manually tested this. It's looking good.

**Review by JessicaSachs — APPROVED** [review]

**Review by JessicaSachs — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**